src/theory/strings/rewrites

```diff
@@ -422,6 +495,100 @@
   (str.contains (str.replace s t r) u)
   true)
 
+(define-cond-rule str-replace-dual-ctn-false ((s ?Seq) (t ?Seq) (r ?Seq) (u ?Seq))
```

Comment:
I am not convinced this holds. What if `s` contains only a part of `u`?

Or even simpler, I think this is a counterexample:
`s` and `r` are empty but `t` and `u` are not. Then, the condition is fulfilled since both contain statements are false. Lets say `t` = "a" and `u`= "b". Then, ` (str.replace t r u)` is  "ba". Since `s` is the empty string it cannot contain "ba".

Reply:
If you case split on whether `t` contains `r`, it is clear to me:
If `t` does not contain `r` then (str.replace t r u) is t, and the equality holds due to the first premise.
If `t` contains `r` then (str.replace t r u) contains `u`, thus the equality holds since `s` does not contain (any string containing) `u` by the second premise.

Note we are concluding negative containment so your counterexample isn't valid.

Reply:
That makes sense, thanks.